packages/beacon-node/src/chain/archiver/archiveStates.ts

```diff
@@ -35,38 +35,49 @@ export class StatesArchiver {
   /**
    * Persist states every some epochs to
    * - Minimize disk space, storing the least states possible
-   * - Minimize the sync progress lost on unexpected crash, storing temp state every few epochs
+   * - Minimize the sync progress lost on unexpected crash, storing last finalized always
    *
    * At epoch `e` there will be states peristed at intervals of `PERSIST_STATE_EVERY_EPOCHS` = 32
    * and one at `PERSIST_TEMP_STATE_EVERY_EPOCHS` = 1024
    * ```
    *        |                |             |           .
    * epoch - 1024*2    epoch - 1024    epoch - 32    epoch
    * ```
+   *
+   * An extra last finalized state is stored which might be cleaned up in next archiving cycle
+   * if there is already a previous state in the `PERSIST_STATE_EVERY_EPOCHS` window
    */
   async maybeArchiveState(finalized: CheckpointWithHex): Promise<void> {
     const lastStoredSlot = await this.db.stateArchive.lastKey();
     const lastStoredEpoch = computeEpochAtSlot(lastStoredSlot ?? 0);
     const {archiveStateEpochFrequency} = this.opts;
 
-    if (finalized.epoch - lastStoredEpoch > Math.min(PERSIST_TEMP_STATE_EVERY_EPOCHS, archiveStateEpochFrequency)) {
-      await this.archiveState(finalized);
+    // If the new big window has started, we need to cleanup on the big interval archiveStateEpochFrequency(1024)
+    // else we need to cleanup within the small PERSIST_TEMP_STATE_EVERY_EPOCHS interval
+    const frequency =
+      Math.floor(lastStoredEpoch / archiveStateEpochFrequency) <
+      Math.floor(finalized.epoch / archiveStateEpochFrequency)
+        ? archiveStateEpochFrequency
+        : PERSIST_TEMP_STATE_EVERY_EPOCHS;
 
-      // Only check the current and previous intervals
-      const minEpoch = Math.max(
-        0,
-        (Math.floor(finalized.epoch / archiveStateEpochFrequency) - 1) * archiveStateEpochFrequency
-      );
+    // Only check the current and previous intervals
+    const minEpoch = Math.max(0, (Math.floor(finalized.epoch / frequency) - 1) * frequency);
```

Comment:
@dapplion @g11tech most of the time `frequency=32` and we store finalized state every 32 epochs which caused the increased DB size, we should use `archiveStateEpochFrequency` here to clean more states to keep stable db size like the old strategy

v1.9.0-rc.0
<img width="1287" alt="Screenshot 2023-05-31 at 14 28 25" src="https://github.com/ChainSafe/lodestar/assets/10568965/955c64e1-1abf-4861-ad3e-02063f069744">

v1.8.0
<img width="1296" alt="Screenshot 2023-05-31 at 14 29 00" src="https://github.com/ChainSafe/lodestar/assets/10568965/93139f8d-c70d-4634-815d-7be0d900108e">